rosbag2_transport/test/rosbag2_transport/test_topic_filter.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <algorithm>
+#include <future>
+#include <map>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "./topic_filter.hpp"
+
+using namespace ::testing;  // NOLINT
+
+TEST(TestTopicFilter, filter_topics_with_more_than_one_type) {
+  std::map<std::string, std::vector<std::string>> topic_with_type;
+  topic_with_type.insert({"topic/a", {"type_a"}});
+  topic_with_type.insert({"topic/b", {"type_b"}});
+  topic_with_type.insert({"topic/c", {"type_c"}});
+  {
+    auto filtered_topics = rosbag2_transport::TopicFilter::filter_topics_with_more_than_one_type(
+      topic_with_type, true);  // include hidden topics
```

Comment:
As I put in another comment - I prefer the style:

```suggestion
      topic_with_type, true /* include_hidden_topics */); 
```

Up to you if you want to follow that convention